components/ApproveCancel/index.tsx

```diff
@@ -34,41 +34,44 @@ export const ApproveCancel = ({
   const wallet = useWallet()
   // @ts-expect-error
   const { serializeParams } = useWasm()
-  const { Address, AvailableBalance } = useMsig()
+  const { Address, AvailableBalance, NumApprovalsThreshold } = useMsig()
 
   // Transaction states
   const [txState, setTxState] = useState<TxState>(TxState.LoadingMessage)
   const [txFee, setTxFee] = useState<FilecoinNumber | null>(null)
 
-  // Get transaction info from url
-  const proposal = getQueryParam.string(router, 'proposal')
-  const approvalsLeft = getQueryParam.number(router, 'approvalsLeft')
-  const transaction = useMemo<MsigTransaction | null>(() => {
+  // Get proposal info from url
+  const proposalString = getQueryParam.string(router, 'proposal')
+  const proposal = useMemo<MsigTransaction | null>(() => {
     try {
-      return JSON.parse(decodeURI(proposal))
+      return JSON.parse(decodeURI(proposalString))
     } catch (e) {
       setTxState(TxState.LoadingFailed)
       return null
     }
-  }, [proposal])
+  }, [proposalString])
+  const approvalsLeft = useMemo<number | null>(() => {
+    if (!NumApprovalsThreshold || !proposal?.approved) return null
+    return NumApprovalsThreshold - proposal.approved.length
+  }, [NumApprovalsThreshold, proposal?.approved])
 
   // Get parameters object to pass to Parameters component
   const parameters = useMemo<Record<string, any> | null>(() => {
-    if (!transaction) return null
-    const { id, approved, proposalHash, ...params } = transaction
+    if (!proposal) return null
+    const { id, approved, proposalHash, ...params } = proposal
```

Comment:
I added `ignoreRestSiblings: true` to the eslint config a while back so that we can use object spread to get a subsection of an object:

```
'@typescript-eslint/no-unused-vars': [
  'error',
  {
    argsIgnorePattern: '^_',
    ignoreRestSiblings: true
  }
]
```